pkg/kubelet/images/image_gc_manager.go

```diff
@@ -240,6 +240,30 @@ func (im *realImageGCManager) GetImageList() ([]container.Image, error) {
 	return im.imageCache.get(), nil
 }
 
+// isContainerActuallyUsingImage determines if a container is actually using its image.
+// Containers in error states or not running are not considered to be using their image
+// for garbage collection purposes.
+func isContainerActuallyUsingImage(container *container.Container) bool {
+	// Only consider containers that are actually running or in a valid state
+	// that indicates they are using the image
+	switch container.State {
+	case "running":
+		return true
+	case "created":
+		// Created containers may be using the image but haven't started yet
+		return true
+	case "exited":
+		// Exited containers are not currently using the image
+		return false
+	case "unknown":
+		// Unknown state containers are not considered to be using the image
+		return false
+	default:
+		// For any other state, be conservative and assume they're not using the image
+		return false
+	}
+}
```

Comment:
nit:
```suggestion
func isContainerActuallyUsingImage(c *container.Container) bool {
	// Only consider containers that are actually running or created
	// that indicates they are using the image
	return c.State == container.ContainerStateRunning || c.State == container.ContainerStateCreated	
}
```

If you prefer to use switch, please consider at least using constants from the container module instead of hardcoded strings.